Update Decidim Analytics with various improvements 0.30

sdelafuente2013 opened the pull request:
- Upgraded Rails dependencies to version 7.0.8.7 and Decidim to ~> 0.30.0. This ensures compatibility with the latest features and fixes.
- Simplified the `Gemfile` by removing outdated constraints and updating gem versions, reducing dependency complexity.
- Enhanced Matomo integration by implementing a helper and adding configuration validation in the controller.
- Upgraded the admin analytics interface with dynamic iframe updates and localized messaging for better usability.
- Included updates to `.gitignore` and introduced `.tool-versions` for improved environment configuration.
- Performed general cleanup, optimizing code throughout the module.

sdelafuente2013 committed:
Refactor analytics module:

- Upgraded Rails version to 7.0.8.7 and Decidim dependencies to ~> 0.30.0.
- Simplified Gemfile by removing unnecessary constraints and updating gems.
- Enhanced Matomo integration: added helper and configuration checks in controller.
- Improved admin analytics UI with dynamic iframe updates and localized messages.
- Updated `.gitignore` and added `.tool-versions`.
- General code cleanup across the module.

alexrlpz reviewed (changes requested):
We should change .ruby-version to use 3.3.4, which is the version required for Decidim 0.30.
You can check it on 0.30 release notes:
https://github.com/decidim/decidim/releases/tag/v0.30.0

fblupi commented on decidim-analytics.gemspec:
```diff
@@ -1,22 +1,30 @@
+  s.authors = ['Pierre Mesure']
+  s.email = ['pierre@mesu.re']
+  s.license = 'AGPL-3.0'
+  s.homepage = 'https://github.com/decidim/decidim-module-analytics'
```
We should change this URL as it does not exist anymore.

fblupi commented on decidim-analytics.gemspec:
```diff
@@ -1,22 +1,30 @@
+  s.license = 'AGPL-3.0'
+  s.homepage = 'https://github.com/decidim/decidim-module-analytics'
+  s.metadata['rubygems_mfa_required'] = 'true'
+  s.required_ruby_version = '>= 3.1.0'
```
If we are upgrading to Decidim 0.30, the required ruby version is different

fblupi commented on decidim-analytics.gemspec:
```diff
@@ -1,22 +1,30 @@
+  s.add_development_dependency 'decidim-dev', Decidim::Analytics::DECIDIM_VERSION
+
+  s.metadata['homepage_uri'] = s.homepage
+  s.metadata['source_code_uri'] = 'https://github.com/decidim/decidim-module-analytics'
```
Like the comment above, these URLs don't exist anymore

fblupi commented on lib/decidim/analytics/admin_engine.rb:
```diff
@@ -2,13 +2,17 @@
+
       routes do
-        resources :analytics, :only => [:index]
+        # Add admin routes here
```
Useless comment

fblupi commented on app/views/decidim/analytics/admin/analytics/index.html.erb:
```diff
@@ -1,18 +1,75 @@
+  </script>
+<% end %>
+
+<style>
```
Why don't you include the styles in the app/packs/stylesheets instead?

fblupi commented on app/controllers/decidim/analytics/admin/analytics_controller.rb:
```diff
@@ -4,11 +4,25 @@ module Decidim
+        private
+
+        def matomo_config
+          @matomo_config ||= Rails.application.credentials.dig(:matomo) ||
```
In decidim we have never used credentials and we won't use it as it makes difficult to manage the instances. Use the secrets instead. Or, even better, the module configuration with env vars. As the secrets won't be available from decidim 0.31

fblupi commented on app/views/decidim/analytics/admin/analytics/index.html.erb:
```diff
@@ -1,18 +1,75 @@
+            <p><%= t("decidim.analytics.admin.configuration_help") %></p>
+
+            <h6><%= t("decidim.analytics.admin.configuration_instructions") %></h6>
+            <pre><code># config/credentials.yml.enc (recomendado)
```
As I told you before, we don't use credentials. We should have the module configuration so we can do something like `Decidim::Analytics.matomo[:server_address]` to access it. And this configuration would take the values from env vars like `DECIDIM_ANALYTICS_MATOMO_SERVER_ADDRESS`. Check it out how decidim handle the configuration of the maps for example. You can configure it by only defining env variables.

Edit: I even see in the README.md we already have the environment variables. So I would just use the existing secrets. No extra work here, just remove all the credentials usage.

fblupi reviewed (changes requested):
We should also include linters for rubocop, erb, js and scss. Ask @apuertadelaguila or @alexrlpz how to do it as they did it before for other modules.

sdelafuente2013 committed:
Refactor Matomo configuration and improve helper methods

- Simplify configuration loading by removing credentials fallback
- Update helper method to accept parameters for better flexibility
- Improve tracking script configuration logic
- Clean up admin engine routing

sdelafuente2013 committed:
Improve analytics dashboard UI and add asset organization

- Simplify iframe configuration and styling approach
- Remove inline styles in favor of external stylesheets
- Add dedicated JavaScript and CSS asset files
- Update configuration instructions to focus on secrets.yml
- Enhance iframe accessibility with proper title attribute

sdelafuente2013 committed:
Update repository metadata and Ruby version requirements

- Update homepage and source code URLs to platoniq organization
- Bump minimum Ruby version requirement to 3.3.4
- Update bug tracker URL for issue reporting

fblupi commented:
We are going to update to the 0.29 before. It will be easier to start the migration to the 0.30 from scratch once  #2 is merged.
